core/shared/src/main/scala/cats/parse/Parser.scala

```diff
@@ -990,12 +1005,23 @@ object Parser {
     * note: this can wind up parsing nothing
     */
   def repAs0[A, B](p1: Parser[A])(implicit acc: Accumulator0[A, B]): Parser0[B] =
-    Impl.Rep0(p1, acc)
+    Impl.Rep0(p1, Int.MaxValue, acc)
+
+  /** Repeat this parser 0 or more times
+    * note: this can wind up parsing nothing
+    */
+  def repAs0[A, B](p1: Parser[A], max: Int)(implicit acc: Accumulator0[A, B]): Parser0[B] =
+    Impl.Rep0(p1, max, acc)
 
   /** Repeat this parser 1 or more times
     */
   def repAs[A, B](p1: Parser[A], min: Int)(implicit acc: Accumulator[A, B]): Parser[B] =
-    Impl.Rep(p1, min, acc)
+    repAs(p1, min, Int.MaxValue)
```

Comment:
Noting: here `Int.MaxValue` means infinity. But below, if a user passes Int.MaxValue they mean literally Int.MaxValue. I get your point that since Strings are indexed by Int, maybe this distinction is academic, but still I'd like to max the distinction just to make the code clearer.

Reply:
I think I understand your point now -- explicitly passed Int.MaxValue for max should never mean no maximum, but just mean Int.MaxValue times (even if the distinction at least for the time being is not observable), so the parsing methods that do the actual loop should represent their max value as max value - 1 to be able to distinguish between looping Int.MaxValue times and looping forever.